controllers/k8ssandra/stargate.go

```diff
@@ -30,7 +30,12 @@ func (r *K8ssandraClusterReconciler) reconcileStargate(
 ) result.ReconcileResult {
 
 	kcKey := client.ObjectKey{Namespace: kc.Namespace, Name: kc.Name}
-	stargateTemplate := dcTemplate.Stargate.Coalesce(kc.Spec.Stargate)
+	stargateTemplate, err := dcTemplate.Stargate.Merge(kc.Spec.Stargate)
```

Comment:
No because the `Merge` method belongs to the `*StargateDatacenterTemplate` type, which is a pointer. This means that if `dcTemplate.Stargate` is nil, then `dcTemplate.Stargate.Merge` will be called on a nil receiver, `in` will be nil, and nil will be returned.

Reply:
I follow you now. I tested [here](https://go.dev/play/p/mnOB2QcOVzo) and got the expected nil pointer dereference. But if I change the type of `Child` to `*Child` it works.

TIL...